src/orchestrator/unary.rs

```diff
@@ -754,7 +762,10 @@ pub async fn detect_for_generation(
     let client = ctx
         .clients
         .get_as::<TextGenerationDetectorClient>(&detector_id)
-        .unwrap();
```

Comment:
FYI - these `clients.get_as::<T>()` calls should be "safe" unwraps as long as the `detector_id` is valid, which we should check prior to this